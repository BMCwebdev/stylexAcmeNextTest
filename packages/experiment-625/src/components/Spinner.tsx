import * as stylex from '@stylexjs/stylex';
import { colorsText } from '@brianAcme/stitch-tokens/colors.stylex';
import { fonts } from '@brianAcme/stitch-tokens/fonts.stylex';
import { Stack } from './layout/Stack';

const styles = stylex.create({
  spinner: {
    width: '48px',
    height: '48px',
    backgroundImage: `url('data:image/svg+xml,<svg width="48" height="48" viewBox="0 0 48 48" color="%23381360" fill="none" xmlns="http://www.w3.org/2000/svg"><defs><linearGradient id="spinner-secondHalf"><stop offset="0%" stop-opacity="0" stop-color="currentColor" /><stop offset="100%" stop-opacity="0.5" stop-color="currentColor" /></linearGradient><linearGradient id="spinner-firstHalf"><stop offset="0%" stop-opacity="1" stop-color="currentColor" /><stop offset="100%" stop-opacity="0.5" stop-color="currentColor" /></linearGradient></defs><g stroke-width="6"><path stroke="url(%23spinner-secondHalf)" d="M 4 24 A 14 14 0 0 1 44 24" /><path stroke="url(%23spinner-firstHalf)" d="M 44 24 A 14 14 0 0 1 4 24" /><!-- 1deg extra path to have the round end cap--><path stroke="currentColor" stroke-linecap="round" d="M 4 24 A 14 14 0 0 1 4 22" /></g><animateTransform from="0 0 0" to="360 0 0" attributeName="transform" type="rotate" repeatCount="indefinite" dur="800ms" /></svg>')`,
  },
  text: {
    color: colorsText.secondary,
    fontSize: fonts.size_200,
    fontWeight: fonts.weight_500,
  },
});

export interface SpinnerProps {
  message: string;
}

export function Spinner({
  message = 'Loading',
}: SpinnerProps) {
  return (
    <Stack horizontalAlign="center" space="200">
      <div {...stylex.props(styles.spinner)}></div>
      <div {...stylex.props(styles.text)}>
        {message}
      </div>
    </Stack>
  );
}
