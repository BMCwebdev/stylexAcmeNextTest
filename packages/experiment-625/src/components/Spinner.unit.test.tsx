import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import { Spinner } from './Spinner';

describe('Spinner', async () => {
  test('renders a basic spinner', async () => {
    const { container } = render(<Spinner></Spinner>);
    expect(screen.getByText('Loading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('renders a spinner with custom message', async () => {
    const { container } = render(<Spinner message="something else" />);
    expect(screen.getByText('something else')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
