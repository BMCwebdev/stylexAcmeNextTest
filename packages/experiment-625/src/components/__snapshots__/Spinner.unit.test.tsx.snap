// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner > renders a basic spinner 1`] = `
<div>
  <div
    class="Stack__stackStyles.base xrvj5dj x3xbxsf x1y6fwsi x193iq5w Stack__alignStyles.start x10ukxgv Stack__justifyItemsStyles.center x1o2pa38 Stack__spaceStyles.200 x15fu49g"
  >
    <div
      class="Spinner__styles.spinner x1useyqa xsdox4t xvzbl3"
    />
    <div
      class="Spinner__styles.text x155a4kq x5g6463 xej1d2r"
    >
      Loading
    </div>
  </div>
</div>
`;

exports[`Spinner > renders a spinner with custom message 1`] = `
<div>
  <div
    class="Stack__stackStyles.base xrvj5dj x3xbxsf x1y6fwsi x193iq5w Stack__alignStyles.start x10ukxgv Stack__justifyItemsStyles.center x1o2pa38 Stack__spaceStyles.200 x15fu49g"
  >
    <div
      class="Spinner__styles.spinner x1useyqa xsdox4t xvzbl3"
    />
    <div
      class="Spinner__styles.text x155a4kq x5g6463 xej1d2r"
    >
      something else
    </div>
  </div>
</div>
`;
